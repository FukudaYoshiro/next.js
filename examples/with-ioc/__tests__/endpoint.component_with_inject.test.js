/* eslint-env jest */

/*
 * Individual component testing is pretty simple
 * just provide your dependencies as props
 * and add `.dive()` step to your shallow render,
 * as with any High Order Component.
 *
 * Remarks about `.html()` may apply,
 * depending if any of the children components
 * expect anything from the context
 */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Component from '../components/endpoint.js'

describe('With Enzyme', () => {
  it('Component renders with props', () => {
    // no need to mock Link component much for shallow rendering
    const injected = shallow(<Component Link={() => {}} />)
    const component = injected.dive()
    expect(component.find('h3').text()).toEqual('Endpoint')
    expect(component.find('Link').first().find('a').text()).toEqual('About: foo baz')
  })
})

describe('With Snapshot Testing', () => {
  it('Blog renders components', () => {
    const component = renderer.create(<Component Link={(props) => <div comment={'mocked Link component'}>{props.children}</div>} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
