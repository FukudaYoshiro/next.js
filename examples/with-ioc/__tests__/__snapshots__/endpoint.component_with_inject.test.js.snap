// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Blog renders components 1`] = `
<div
  style={
    Object {
      "border": "1px dashed #00ff00",
      "marginTop": "5px",
      "padding": "10px",
    }
  }
>
  <h3>
    Endpoint
  </h3>
  Uses injected \`Link\` component without direct dependency on one
  <br />
  <div
    comment="mocked Link component"
  >
    <a>
      About: foo baz
    </a>
  </div>
  <br />
  <div
    comment="mocked Link component"
  >
    <a>
      go Home
    </a>
  </div>
</div>
`;
